ssm_jax/ekf/inference.py

```diff
@@ -0,0 +1,195 @@
+import jax.numpy as jnp
+import jax.random as jr
+from jax import lax
+from jax import jacfwd
+from distrax import MultivariateNormalFullCovariance as MVN
+from ssm_jax.nlgssm.containers import NLGSSMPosterior
+
+
+# Helper functions
+_get_params = lambda x, dim, t: x[t] if x.ndim == dim+1 else x
+
+def _predict(m, S, f, F, Q, u):
+    """Predict next mean and covariance using first-order additive EKF
+
+        p(x_{t+1}) = \int N(x_t | m, S) N(x_{t+1} | f(x_t, u), Q)
+                    = N(x_{t+1} | f(m, u), F(m, u) S F(m, u)^T + Q)
+
+    Args:
+        m (D_hid,): prior mean.
+        S (D_hid,D_hid): prior covariance.
+        f (Callable): dynamics function.
+        F (Callable): Jacobian of dynamics function.
+        Q (D_hid,D_hid): dynamics covariance matrix.
+        u (D_in,): inputs.
+
+    Returns:
+        mu_pred (D_hid,): predicted mean.
+        Sigma_pred (D_hid,D_hid): predicted covariance.
+    """
+    F_x = F(m, u)
+    mu_pred = f(m, u)
+    Sigma_pred = F_x @ S @ F_x.T + Q
+    return mu_pred, Sigma_pred
+
+
+def _condition_on(m, S, h, H, R, u, y):
```

Comment:
I suggest we follow Sarkka's notation and use P for the prior covariance,
so we can define the intermediate variable `S_t = H_t Sigma_{t|t-1} H_t' + R_t` and hence
`K = Sigma_{t|t-1} H_t S_t^{-1}`. 
Also, can we just reuse the following?
https://github.com/probml/ssm-jax/blob/main/ssm_jax/lgssm/inference.py#L77

Reply:
Sounds good!